src/storage/src/store.rs

```diff
@@ -242,7 +243,13 @@ pub trait LocalStateStore: StaticSendSync {
 
     fn is_dirty(&self) -> bool;
 
-    fn init(&mut self, epoch: u64);
+    /// Initializes the state store with given `epoch` pair.
+    /// Typically we will use `epoch.curr` as the initialized epoch,
+    /// Since state table will begin as empty.
+    /// In some cases like replicated state table, state table may not be empty initially,
+    /// as such we need to wait for `epoch.prev` checkpoint to complete,
+    /// hence this interface is made async.
+    fn init(&mut self, epoch: EpochPair) -> impl Future<Output = StorageResult<()>> + Send + '_;
```

Comment:
I think we can have an `InitOptions` rather than a raw epoch pair in case that in the future we want to pass more fields in `init`.